src/backend/distributed/commands/multi_copy.c

```diff
@@ -3465,17 +3557,50 @@ CopyGetPlacementConnection(ShardPlacement *placement, bool stopOnFailure)
 	}
```

Comment:
is it ok that we're not calling `MarkRemoteTransactionCritical` here?

Reply:
I want to use `MarkRemoteTransactionCritical` along with `ClaimConnectionExclusively`.

Running some more tests for making sure consecutive COPY queries do not try to claim a connection that is not unclaimed by the previous one

Reply:
I think the main effect of not calling ClaimConnectionExclusively would be that, if a connection already wrote to a shard and the shard is the first shard that the COPY writes to, we might start using that connection for multiple shard , instead of opening separate connections per shard.

Reply:
There's a more subtle effect that is worth documenting in a comment: 

TODO: We do not claim the connection exclusively, because may need to be returned by GetConnectionIfPlacementAccessedInXact multiple times and claiming exclusively prevents that. That means we are not guarded against distributed queries that happen during the COPY (e.g. a distributed query in a function that is called to compute a default value). The proper solution would be to claim all connections which already modified shard placements at the start of the COPY, regardless of whether rows are sent to those shard placements during the COPY.